components/project-list/project-list.jsx

```diff
@@ -16,28 +16,48 @@ export default React.createClass({
       displayBatchIndex: 1
     };
   },
+  getDefaultProps() {
+    return {
+      params: {
+        featured: null,
+        issue: null,
+        search: null,
+        ids: null
+      }
+    };
+  },
   componentDidMount() {
     this.fetchData(this.props.params);
   },
   componentWillReceiveProps(nextProps) {
+    // Reset state
+    this.resetState();
```

Comment:
Since we are doing search-as-you-type (we initiate the search when there's no activity happening in the input box for x amount of time), for me having the previous search result stayed while new search is happening in the background can confuse me. If there's no UI change go along with it, as an user I won't know a new search has already been initiated. Having a spinner instead of text `Searching for entries that match ‘some_keyword’...` sounds great! But I do think we should clear previous search result while fetching results from Pulse API for the current search keyword.

@alanmoo @kristinashu thoughts? This is what has been implemented in this PR
![pulse search 2](https://cloud.githubusercontent.com/assets/2896608/22705133/359ffcf2-ed1f-11e6-916b-b95b56ce9926.gif)